code/processes/gateway.q

```diff
@@ -412,13 +413,13 @@ getserveridstype:{[att;typ]
 	getserverscross[att;servers;besteffort]];
 
   serverids:first value flip $[99h=type res; key res; res];
-  if[all 0=count each serverids; '"no servers match ",string[typ]," requested attributes"];
+  if[all 0=count each serverids;.gw.formatresponse[0b;`sync;"no servers match ",string[typ]," requested attributes"]];
   :serverids;
  }
 
 // execute an asynchronous query
 asyncexecjpt:{[query;servertype;joinfunction;postback;timeout]
- if[.gw.permissioned;if[.pm.allowed[.z.u; query];'"User is not permissioned to run this query from the gateway"]];
+ if[.gw.permissioned;if[.pm.allowed[.z.u; query];.gw.formatresponse[0b;`sync;"User is not permissioned to run this query from the gateway"]]];
```

Comment:
I don't think this is correct - I think this is a bug in the original.  This should also be propagated back via the neg handle